Sources/PostgreSQL/Connection/PostgreSQLConnection+NotifyAndListen.swift

```diff
@@ -0,0 +1,29 @@
+import Async
+
+extension PostgreSQLConnection {
+    public func listen(
+        _ channelName: String,
+        handler: @escaping (String) throws -> ()
+        ) throws -> Future<Void> {
+        beforeClose = { conn in
```

Comment:
Sorry, but what would happen if we try to listen to more than one channel? Wouldn’t the first one never get unlistened then? Or is it a bad idea to use a connection that listens to one specific channel for anything else, anyway?

If calling `listen` means that this connection should no longer be used for anything else, I think that should be clearly documented.

(Technically, there should be little reason not to listen to more than one channel on the same connection (they can be switched on via process id/channel name), but I understand that that might be infeasible given the way `QueueHandler` works at the moment. Still worth considering for the future, though, given that Postgres has a somewhat low connection limit (100) by default.)

Reply:
I welcome a PR to enhance this with all your helpful suggestions! Each Listen command has to be executed individually so the only modification needed should be to add "unlisten" statements. https://www.postgresql.org/docs/10/static/sql-listen.html you can read all about the technicals here. That would be a great enhanement.

Reply:
I guess I can make this happen it really shouldnt be so bad, its really just the cleanup thats kind of more then I expected .. .meh no big deal!

Reply:
I think that is like asking what happens if I make a query and it does not execute because of a query heppens then? If this were GRPC or AMQP I would totally agree with you that there should be a required failurehandler. but this is not a designated message broker. This seems like a good enahncement but not at this level this should be more at the top level of the connection.... which currently if there is a failure you can "catch" the error by virtue of the way eventloopfutures work. If you want something water proof it is beyond the scope of this PR as that would be an upstream change

Reply:
But how is this supposed to be used if I can't rely on getting notified, because the connection might close at any time without notice? I agree that this might not be part of this PR, but I think it should arrive least be clear what happens/what to do if the listening connection closes.

Reply:
same question... what happens if the connection to PostgreSQL fails not if a notify/Listen handler fails. if you want to open a separate issue I am sure you myself or one of the many other members of the community can work on this. This is still Out of scope